proxy/src/fs_watch.rs

```diff
@@ -373,6 +394,67 @@ mod tests {
         });
     }
 
+    fn test_nonexistent_files_dont_file_delete_events(
+        fixture: Fixture,
+        watch: Watch<()>,
+        bg: Box<Future<Item = (), Error = ()>>,
+    ) {
+        // This test confirms that when a file has been deleted,
+        // it's nonexistence doesn't continuously generate new
+        // "file deleted" events.
+        let Fixture {
+            paths,
+            dir: _dir,
+            mut rt,
+        } = fixture;
+
+        rt.spawn(bg);
+
+        let watch = paths.iter().fold(watch, |watch, ref path| {
+            create_and_write(path, b"A").unwrap();
+
+            let (item, watch) = next_change(&mut rt, watch).unwrap();
+            assert!(item.is_some());
+            watch
+        });
+
+        let watch = paths.iter().fold(watch, |watch, ref path| {
+            fs::remove_file(path).unwrap();
+            println!("deleted {:?}", path);
+
+            let (item, watch) = next_change(&mut rt, watch).unwrap();
+            assert!(item.is_some());
+            watch
+        });
+
+        let watch2 = watch.clone();
+        // Check if the stream has become ready every one second.
+        let stream = ::tokio::timer::Interval::new(
+            Instant::now(),
+            Duration::from_secs(1),
+        )
+            .map(|_| {
```

Comment:
pretty jarring. perhaps create the interval in its own statement.  Also it's a red flag to see `Instant::now()`.  Shouldn't we be using tokio's clock?